### PR TITLE
Add processes endpoint

### PR DESCRIPTION
**Pull request by geospatial-roman**
Relates to Issue #6 

Adds /processes endpoint to query availabled openeo Processes from openeo-processes dask

### CODE REVIEW

File: openeo_fastapi/client/collections.py [hunk 1]
```diff
@@ -13,27 +11,61 @@ async def get_collections():
...
+        async with aiohttp.ClientSession() as client:
+            async with client.get(stac_url + "collections") as response:
+                resp = await response.json()
+                if response.status == 200 and resp.get("collections"):
+                    collections_list = []
+                    for collection_json in resp["collections"]:
+                        # For the collections from STAC, only let them through if they're on the whitelist
+                        # This has to be before the legacy collections are added.
+                        if (
+                            len(app_settings.STAC_COLLECTIONS_WHITELIST) < 1
```
Comment (SerRichard) [inline openeo_fastapi/client/collections.py:30]: Empty lists evaulate as false, so tidier would be
```suggestion
                           not app_settings.STAC_COLLECTIONS_WHITELIST
```

File: openeo_fastapi/client/collections.py [hunk 2]
```diff
@@ -13,27 +11,61 @@ async def get_collections():
...
+    )
+
+    try:
+        async with aiohttp.ClientSession() as client:
+            async with client.get(stac_url + "collections") as response:
+                resp = await response.json()
+                if response.status == 200 and resp.get("collections"):
+                    collections_list = []
+                    for collection_json in resp["collections"]:
+                        # For the collections from STAC, only let them through if they're on the whitelist
```
Comment (SerRichard) [inline openeo_fastapi/client/collections.py:27]: Remove eodc code comment lines :D Not sure how to suggest that in line?
```suggestion
```

File: openeo_fastapi/client/collections.py [hunk 3]
```diff
@@ -13,27 +11,61 @@ async def get_collections():
...
+                        # This has to be before the legacy collections are added.
```
Comment (SerRichard) [inline openeo_fastapi/client/collections.py:28]:
```suggestion
```

File: openeo_fastapi/client/collections.py [hunk 4]
```diff
@@ -13,27 +11,61 @@ async def get_collections():
...
+
+    try:
+        async with aiohttp.ClientSession() as client:
+            async with client.get(
+                stac_url + f"collections/{collection_id}"
+            ) as response:
+                resp = await response.json()
+                if response.status == 200 and resp.get("id"):
+                    if (
+                        len(app_settings.STAC_COLLECTIONS_WHITELIST) < 1
```
Comment (SerRichard) [inline openeo_fastapi/client/collections.py:63]: same as above ( I see this is copies from out bit of code, so the git blame is with us ;) )

File: .devcontainer/docker-compose.yml
Comment (SerRichard) [inline .devcontainer/docker-compose.yml:1]: If you are using this, then I won't merge the removal PR?

**Comment by SerRichard** [comment]
Yeah, minor comments, otherwise lgtm!

**Review by SerRichard — COMMENTED** [review]
I was about to say add tests for collections, and then I realised I was reading the processes MR. Don't know if it was intentional or accidental to mix the two branches? :)

File: .devcontainer/docker-compose.yml
Comment (geospatial-roman) [reply .devcontainer/docker-compose.yml:1]: As discussed, let's keep it until we find a better solution

**Review by SerRichard — APPROVED** [review]
Some of the dependencies might not be needed later as it's just an api implementation, i.e xvec for example, but lgtm and can be ignored for now

**Pull request merged**